packages/vm/src/runBlock.ts

```diff
@@ -285,6 +285,12 @@ export async function runBlock(this: VM, opts: RunBlockOpts): Promise<RunBlockRe
     preimages: result.preimages,
   }
 
+  if (this.common.isActivatedEIP(7685)) {
```

Comment:
we can do request field existence check even before we run the block

Also we need to pop requests out before we verify stateroot (and call verkle verify state)